src/NovelRT/Audio/AudioService.cpp

```diff
@@ -334,6 +336,13 @@ AudioService::~AudioService() {
   alDeleteSources(1, &_soundSource);
   alDeleteSources(1, &_musicSource);
 
+  for (auto it = _sounds.begin(); it != _sounds.end(); it++) {
+    unload(it, false);
+  }
+  for (auto it = _music.begin(); it != _music.end(); it++) {
+    unload(it, true);
+  }
+
```

Comment:
Is this something that we should keep? From testing, it appears that something is making OpenAL automatically delete the buffers that are created, but its not sure it's the dtor since it seems like the dtor is never being called outright. Unless the logging service is killed prior to OpenAL destruction, the logging calls that were introduced to delete the buffers is never called, hence me thinking that this dtor isn't called.

If it's unnecessary because the dtor is implicitly called elsewhere, or because having the lazy context get destroyed destroys the buffers, I'll remove this since it's redundant.

Reply:
Assuming they are needed - these should most likely be `for(auto sound : _sounds)` and `for(auto musicObject : _music)` accordingly.

Alternatively, you can use `std::foreach`.